[SCRUM-494] Add onboarding page and integrate backend API

Vincent23412 opened the pull request:
<!-- markdownlint-disable MD041 -->

## Description

This PR implements the frontend onboarding page as required by the related Jira issue.

A new onboarding page is added under `/onboarding/Vincent`, which integrates with the backend onboarding API to fetch and display self-introduction information.  
The implementation follows the **TPET Frontend Coding Guidelines**, with the UI kept flexible and simple.

<!--
Provide a brief description of this PR:
- What issue does this PR address?
- Why is this change necessary?
- How was it implemented?
-->

---

## Type of Change

<!--
Select the type(s) of changes made (you can select multiple):
-->

- [ ] 🐛 Bug Fix (fixes an issue)
- [x] ✨ New Feature (introduces a new feature)
- [x] 💄 UI/UX (improves interface or user experience)
- [ ] 🔨 Refactor (code refactoring)
- [ ] 📝 Documentation (updates documentation)
- [ ] 🔧 Configuration (changes configuration)
- [ ] 🚀 Performance (improves performance)
- [ ] ✅ Test (related to tests)
- [ ] 🔒 Security (addresses security concerns)

---

## Related Issues

Relates to [SCRUM-511](https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-511?atlOrigin=eyJpIjoiMzQ1YjYwY2RkODllNDdjMzg1NzdhMzljYjlkZTllN2QiLCJwIjoiaiJ9)

<!--
List related issue numbers or links:
Example: Closes SCRUM-123, Relates to SCRUM-456
-->

---

## Testing

### Page Route

[https://deploy-preview-93--tpet-dev-aws-educate.netlify.app/onboarding/vincent](https://deploy-preview-93--tpet-dev-aws-educate.netlify.app/onboarding/vincent)


1. Start the frontend application locally
2. Open the following URL in the browser:
3. Verify that:
- The page renders successfully
- The frontend sends a request to the backend onboarding API
- The self-introduction information is displayed correctly on the UI

Backend API specification:
- https://documenter.getpostman.com/view/36968695/2sB3dTt8Ro

<!--
Describe how these changes were tested:
- What tests were performed?
- How were these changes verified?
- Are there any specific test steps required?
-->

---

## Screenshots/Videos

<img width="1439" height="815" alt="截圖 2025-12-27 下午2 15 34" src="https://github.com/user-attachments/assets/a3dc61ca-2606-4b32-a294-ff9e4e47f527" />

---

## Checklist

<!--
Ensure the following items are checked before submitting the PR:
-->

- [x] I have tested these changes.
- [ ] I have updated the relevant documentation.
- [x] My code adheres to the project’s code standards.
- [x] I have addressed all console warnings/errors.
- [x] I have removed all `console.log` or `print()` statements.
- [ ] I have added necessary test cases.
- [x] All existing tests pass.

---

## Additional Notes

N/A

<!--
Reminder:
Before submitting your PR, please review the **Coding Guidelines**:
https://aws-educate-tw.notion.site/TPET-Backend-Coding-Guidelines-12e6bfee681780ac89c3cf43854380d4?pvs=4
-->

<!-- markdownlint-enable MD041 -->


[SCRUM-511]: https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-511?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Vincent23412 committed:
feat: add onboarding/vincent page with API integration | SCRUM-511

ki225 commented:
@Vincent23412 

Could you also add the page URL (route) in the description or the Testing section?
For example: `<base_url>/home`.
We couldn’t find it through the UI components.

Also, please attach the related Jira issue link.

Thanks!

Vincent23412 commented:
Thanks for the suggestion!

I’ve updated the PR to explicitly include the page route (`http://localhost:3000/onboarding/Vincent`) in the Testing section and added the related Jira issue link (SCRUM-511).

Please let me know if anything else needs adjustment. Thanks!

ki225 commented on src/app/onboarding/vincent/page.tsx:
```diff
@@ -0,0 +1,63 @@
+};
+
+const onboardingApiUrl =
+  "https://1ytzswuegc.execute-api.us-west-2.amazonaws.com/onboarding/vincent";
```
We do not hardcode URLs in the codebase. Please reference `NEXT_PUBLIC_API_ENDPOINT` from the `.env` file instead. For usage details, you can refer to this file as an example: [/src/app/emailHistory/page.tsx#L95](https://github.com/aws-educate-tw/email-sender-nextjs/blob/f484fbfb94b13cb0d0583fefb353f8fcd0172191/src/app/emailHistory/page.tsx#L95)

Before this endpoint can work correctly, please make sure your API is deployed to the **Dev environment**.
You can manually deploy it by following the setup in this PR: [Backend PR #130](https://github.com/aws-educate-tw/aws-educate-tpet-backend/pull/130)

Vincent23412 replied:
Updated to build the URL from NEXT_PUBLIC_API_ENDPOINT (following page.tsx (line 95)) and removed the hardcoded value.

Vincent23412 committed:
chore: build onboarding API url from env

Vincent23412 committed:
chore: fix formatting for CI

chungchihhan commented on src/app/onboarding/vincent/page.tsx:
```diff
@@ -0,0 +1,64 @@
+const onboardingApiUrl = `${apiBaseUrl.replace(/\/$/, "")}/onboarding/vincent`;
+
+async function fetchOnboarding(): Promise<OnboardingResponse> {
+  const response = await fetch(onboardingApiUrl, { cache: "no-store" });
```
What is the purpose of using cache: "no-store" here? Usuaslly we use that for dynamic data.

Vincent23412 replied:
Removed cache: "no-store" and use Next.js default caching since this data is static.

chungchihhan reviewed (approved):
Everythings look great!

Vincent23412 committed:
chore: remove no-store cache from onboarding fetch

chungchihhan reviewed (approved):
Looks great to me

poyang1024 commented on src/app/onboarding/vincent/page.tsx:
```diff
@@ -0,0 +1,64 @@
+};
+
+const apiBaseUrl =
+  process.env.NEXT_PUBLIC_API_ENDPOINT ?? "https://1ytzswuegc.execute-api.us-west-2.amazonaws.com";
```
Hi @Vincent23412 ,

Per Said (@ki225 ), let's set apiBaseUrl using `NEXT_PUBLIC_API_ENDPOINT` from .env.

I considered keeping the hardcoded URL as a fallback for error handling, but since we are pushing this to main, we should strictly remove the hardcoded part to keep the code clean.

Vincent23412 replied:
Updated to build apiBaseUrl from NEXT_PUBLIC_API_ENDPOINT per your guidance and removed the hardcoded fallback to keep production code clean.

Vincent23412 committed:
chore: remove hardcoded onboarding API url
